webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -1233,6 +1268,123 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
             zipOut.putNextEntry(null, parameters);
             zipOut.write(join(stripNewlines(scrambleRequest.getAllScrambles()), "\r\n").getBytes());
             zipOut.closeEntry();
+            
+            // i18n is only for fmc
+            if (!scrambleRequest.fmc){
+                continue;
+            }
+            
+            ArrayList<String> availableLanguages = new ArrayList<String>();
+            // TODO
+            // Jeremy, this is where the iteration over files can let internationalization easier to handle.
+            // The following lines must be replaced.
+            availableLanguages.add("MessagesBundle.properties");
+            availableLanguages.add("MessagesBundle_en_US.properties");
+            availableLanguages.add("MessagesBundle_pt_BR.properties");
+            
+            for (String item : availableLanguages) {
+            
+                String[] temp = item.split("_|\\.");
+                
+                if (temp.length != 4){
+                    continue;
+                }
+                
+                String language = temp[1];
+                String country = temp[2];
+                Locale locale = new Locale(language, country);
+                
+                // fewest moves regular sheet
+                pdfFileName = "pdf/translations/"+language+"_"+country+"_"+safeTitle+".pdf";
+                parameters.setFileNameInZip(pdfFileName);
+                zipOut.putNextEntry(null, parameters);
+
+                ByteArrayOutputStream pdfOut = new ByteArrayOutputStream();
+                Rectangle pageSize = PageSize.LETTER;
+                Document doc = new Document(pageSize, 0, 0, 75, 75);
+                PdfWriter docWriter = PdfWriter.getInstance(doc, pdfOut);
+
+                docWriter.setBoxSize("art", new Rectangle(36, 54, pageSize.getWidth()-36, pageSize.getHeight()-54));
+
+                doc.addCreationDate();
+                doc.addProducer();
+                if(globalTitle != null) {
+                    doc.addTitle(globalTitle);
+                }
+
+                doc.open();
+                for (int i=0; i<scrambleRequest.scrambles.length; i++){
+                    addFmcSolutionSheet(docWriter, doc, scrambleRequest, globalTitle, i, locale);
+                }
+                doc.close();
+
+                pdfReader = new PdfReader(pdfOut.toByteArray());
+                b = new byte[(int) pdfReader.getFileLength()];
+                pdfReader.getSafeFile().readFully(b);
+                zipOut.write(b);
+
+                zipOut.closeEntry();
+                
+                // Cutout sheet. Is this really necessary?
```

Comment:
I agree, this doesn't seem to be necessary, since it's not language specific.

Reply:
Removed